src/vstest.console/Processors/ListFullyQualifiedTestsArgumentProcessor.cs

```diff
@@ -0,0 +1,459 @@
+        /// </summary>
+        internal IOutput output;
+
+        /// <summary>
+        /// RunSettingsManager to get currently active run settings.
+        /// </summary>
+        private IRunSettingsProvider runSettingsManager;
+
+        /// <summary>
+        /// Registers for discovery events during discovery
+        /// </summary>
+        private ITestDiscoveryEventsRegistrar discoveryEventsRegistrar;
+
+        /// <summary>
+        /// Test case filter instance
+        /// </summary>
+        private TestCaseFilter testCasefilter;
+
+        /// <summary>
+        /// List to store the discovered tests
+        /// </summary>
+        private List<string> discoveredTests = new List<string>();
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        /// <param name="options">
+        /// The options.
+        /// </param>
+        public ListFullyQualifiedTestsArgumentExecutor(
+            CommandLineOptions options,
+            IRunSettingsProvider runSettingsProvider,
+            ITestRequestManager testRequestManager) :
+                this(options, runSettingsProvider, testRequestManager, ConsoleOutput.Instance)
+        {
+        }
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        /// <param name="options">
+        /// The options.
+        /// </param>
+        internal ListFullyQualifiedTestsArgumentExecutor(
+            CommandLineOptions options,
+            IRunSettingsProvider runSettingsProvider,
+            ITestRequestManager testRequestManager,
+            IOutput output)
+        {
+            Contract.Requires(options != null);
+
+            this.commandLineOptions = options;
+            this.output = output;
+            this.testRequestManager = testRequestManager;
+
+            this.runSettingsManager = runSettingsProvider;
+            this.testCasefilter = new TestCaseFilter();            
+            this.discoveryEventsRegistrar = new DiscoveryEventsRegistrar(output, this.testCasefilter, discoveredTests, this.commandLineOptions);
+        }
+
+        #endregion
+
+        #region IArgumentExecutor
+
+        /// <summary>
+        /// Initializes with the argument that was provided with the command.
+        /// </summary>
+        /// <param name="argument">Argument that was provided with the command.</param>
+        public void Initialize(string argument)
+        {
+            if (!string.IsNullOrWhiteSpace(argument))
+            {
+                this.commandLineOptions.AddSource(argument);
+            }
+        }
+
+        /// <summary>
+        /// Lists out the available discoverers.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        public ArgumentProcessorResult Execute()
+        {
+            Contract.Assert(this.output != null);
```

Comment:
What happens if user gives /ListTests and /ListFullyQualifiedTests both ?

Reply:
Both will get executed. Also a user is not supposed to know about ListFullyQualifiedTests. This is for internal consumption only.

Reply:
@prawalagarwal Why is this exactly?
As a user, I find it useful to discover fully qualified names of tests, especially in C++ DLLs.

Is there an alternative way to do this?

Reply:
@JunielKatarn We needed this for a feature we introduced back then called "Test Impact". Listing fully qualified tests was not in the PM spec for vstest console options initially. If you have a requirement for the same, you can reach out to devops_tools@microsoft.com and our PMs will take this up.